crates/sdl-encoder/src/field.rs

```diff
@@ -0,0 +1,114 @@
+use crate::FieldType;
+use std::fmt::{self, Display};
+
+/// Field in a given SDL type.
+#[derive(Debug, PartialEq, Clone)]
+pub struct Field {
+    description: Option<String>,
+    //TODO(@lrlna): fields for input objects can also take arguments. This
+    //struct should also account for that.
+    name: String,
+    type_: FieldType,
+}
+
+impl Field {
+    /// Create a new instance of Field.
+    pub fn new(name: String, type_: FieldType) -> Self {
+        Self {
+            description: None,
+            name,
+            type_,
+        }
+    }
+
+    /// Set the field's description.
+    pub fn description(&mut self, description: Option<String>) {
+        self.description = description;
+    }
+}
+
+impl Display for Field {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if let Some(description) = &self.description {
+            // Let's indent description on a field level for now, as all fields
+            // are always on the same level and are indented by 2 spaces.
+            writeln!(f, "  \"\"\"\n  {}\n  \"\"\"", description)?;
+        }
+        // TODO(@lrlna): double check with folks if it's a valid SDL if the last
```

Comment:
Commas are valid everywhere, and treated as whitespace. That means there is no reason to use commas either (as long as there is another whitespace separator between tokens).